src/indexes.rs

```diff
@@ -396,11 +361,11 @@ impl Index {
         } else {
             format!("{}/indexes/{}/documents", self.client.host, self.uid)
         };
-        Ok(request::<&[T], Task>(&url, &self.client.api_key, Method::Post(documents), 202).await?)
+        request::<&[T], Task>(&url, &self.client.api_key, Method::Post(documents), 202).await
     }
 
     /// Alias for [Index::add_or_replace].
-    pub async fn add_documents<T: Document>(
+    pub async fn add_documents<T: DeserializeOwned + Serialize + std::fmt::Debug>(
```

Comment:
The DeserializeOwned is not necessary here. Remove it so that we can add docs like this:
```
  #[derive(Serialize, Debug)]
  struct Movie<'a> {
      name: &'a str,
      description: &'a str,
  }
```

Reply:
Hey @kvinwang, just so you have the whole context over this PR.
There was a little discussion on our slack community a few days ago where @honne23 explained to us why the Owned version was needed;
![image](https://user-images.githubusercontent.com/7032172/163993014-c74b6fc0-6488-4f41-b2ee-fafa68cecbb6.png)

Reply:
> There was a little discussion on our slack community a few days ago where @honne23 explained to us why the Owned version was needed;

The issue here is not about Owned or not. The `add_documents` only need to serialize the data, So neither `DeserializeOwned` or `Deserialize` is needed here.

Reply:
Ah yes you're right my bad!